app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
```

Comment:
Là method tác dụng cho phép lưu password được bảo mật dưới dạng password_digest ở database. Ngoài ra còn tạo thêm các method liên quan như authenticate để xác nhận mật khẩu, attribute password and password_confirmation